### PR TITLE
Menu system

### PR DESCRIPTION
**Pull request by rift137**
WCMS-523 Added Menu.
Menu Links will be another pull request to keep the size down.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ gem 'biola_deploy'
 gem 'biola_frontend_toolkit', '~> 0.4.4'
 gem 'biola_wcms_components', '~> 0.15.0'
 gem 'blazing'
-gem 'buweb_content_models', '~> 0.126.0'
+gem 'buweb_content_models', '~> 0.127.0'
```
Comment (halloffame) [inline Gemfile]: Use 0.127.1

File: app/controllers/menus_controller.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    @menu.page_editions = [] # allows for unsetting of all previously set page_editions
```
Comment (halloffame) [inline app/controllers/menus_controller.rb]: For some reason I can't set page_editions on save

File: app/controllers/menus_controller.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  def show
+    redirect_to [:edit, @menu]
+  end
+
+  def new
+  end
+
+  def create
+    if @menu.save
```
Comment (halloffame) [inline app/controllers/menus_controller.rb]: For some reason I can't set page_editions on create

Comment (rift137) [inline app/controllers/menus_controller.rb]: This issue is resolved now. 
Ref: https://github.com/biola/buweb-content-models/commit/a0db4bd8df3f751ca53687058715c18075196fe8

**Pull request merged**